indico/modules/events/contributions/models/contributions.py

```diff
@@ -47,8 +48,31 @@ def _get_next_friendly_id(context):
     return increment_and_get(Event._last_friendly_contribution_id, Event.id == event_id)
 
 
+class CustomFieldsMixin:
+    """Store information about contribution custom fields."""
+
+    def save_custom_fields(self, custom_field_values):
+        for custom_field in self.event_new.contribution_fields:
+            data = custom_field_values.get('custom_{}'.format(custom_field.id))
+            if data:
+                ContributionFieldValue(contribution_field=custom_field, contribution=self, data=data)
+
+    def set_custom_field(self, field_id, field_value):
+        field = self.get_field(field_id)
+        if field:
```

Comment:
field_value_cls = type(self).field_values.prop.mapper.class_

That way it'll still work once we use it for abstracts since you'll get the proper class in any case